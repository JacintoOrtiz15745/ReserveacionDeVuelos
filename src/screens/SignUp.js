import React, {useState} from 'react';
import {View} from 'react-native';
import {styles} from '../styles/SignUpStyles';
import Form from '../components/molecules/SignUpMolecules/SignUpForm';
import SignUpTitle from '../components/atoms/SignUpAtoms/SignUpTitle';
import SignUpButtons from '../components/atoms/SignUpAtoms/SignUpButtons';
import {validateInputs} from '../utils/Functions';
const SignUp = ({navigation}) => {
  const [formValues, setFormValues] = useState({
    name: '',
    email: '',
    password: '',
    termsCheckbox: false,
    updatesCheckbox: false,
    validName: true,
    validEmail: true,
    validPassword: true,
    validatedName: false,
    validatedEmail: false,
    validatedPassword: false,
  });
  const validationObject = {
    terms: formValues.termsCheckbox,
    updates: formValues.updatesCheckbox,
    validatedName: formValues.validatedName,
    validatedEmail: formValues.validatedEmail,
    validatedPassword: formValues.validatedPassword,
  };
  return (
    <View style={styles.SignUpContainer} testID="SignUpScreen">
      <SignUpTitle />
      <Form formValues={formValues} setFormValues={setFormValues} />
      <SignUpButtons
        validateInputs={validateInputs}
        validationValues={validationObject}
        formValues={formValues}
        navigation={navigation}
      />
    </View>
  );
};

export default SignUp;
