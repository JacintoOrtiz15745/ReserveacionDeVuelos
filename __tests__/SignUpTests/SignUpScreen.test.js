import React from 'react';
import SignUp from '../../src/screens/SignUp';
import {shallow} from 'enzyme';
let wrapper;
beforeEach(() => {
  wrapper = shallow(<SignUp />);
});
describe('renders correctly', () => {
  test('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('renders component', () => {
    const component = wrapper.findWhere(
      (node) => node.prop('testID') === 'SignUpScreen',
    );
    expect(component.exists()).toBeTruthy();
  });
});
