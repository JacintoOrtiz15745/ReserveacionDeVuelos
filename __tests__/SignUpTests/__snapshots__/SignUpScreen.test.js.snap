// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly snapshot 1`] = `
<View
  style={
    Object {
      "margin": 10,
      "padding": 10,
    }
  }
  testID="SignUpScreen"
>
  <SignUpTitle />
  <SignUpForm
    formValues={
      Object {
        "email": "",
        "name": "",
        "password": "",
        "termsCheckbox": false,
        "updatesCheckbox": false,
        "validEmail": true,
        "validName": true,
        "validPassword": true,
        "validatedEmail": false,
        "validatedName": false,
        "validatedPassword": false,
      }
    }
    setFormValues={[Function]}
  />
  <SignUpButtons
    formValues={
      Object {
        "email": "",
        "name": "",
        "password": "",
        "termsCheckbox": false,
        "updatesCheckbox": false,
        "validEmail": true,
        "validName": true,
        "validPassword": true,
        "validatedEmail": false,
        "validatedName": false,
        "validatedPassword": false,
      }
    }
    validateInputs={[Function]}
    validationValues={
      Object {
        "terms": false,
        "updates": false,
        "validatedEmail": false,
        "validatedName": false,
        "validatedPassword": false,
      }
    }
  />
</View>
`;
